### ISSUE TITLE
[regression/7.0.59] Can't set MAUI iOS Navigation Icon after updating to 17.6 Preview 1

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/maui#13914]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Cant-set-MAUI-iOS-Navigation-Icon-after/10305690)._

---
[severity:I'm unable to use this version] [regression] [worked-in:17.5 Preview 6]
I am using an iOS NavigationRenderer to replace the leftmost icon on the NavigationBar in a NavigationPage.
This worked perfectly with 17.5 Preview 6, but with 17.6 Preview 1 it only works on pages that are not the top page.
Instead of the custom icon (looks like Android hamburger icon), I get the word "Menu".
Also, the UIBarButtonItem event handler only fires when the page is not the top page.

![ioshamburegerbutton.png](https://aka.ms/dc/image?name=Bbf3abeb0a85649f2ac5b1af4b7d01818638139707163390427_ioshamburegerbutton.png&tid=bf3abeb0a85649f2ac5b1af4b7d01818638139707163390427)

Here is the code from the NavigationRenderer:
```
protected override Task OnPushAsync(Page page, bool animated)
{
    var pushed = base.OnPushAsync(page, animated);

    if (TopViewController != null)
    {
        var hasBackButton = page.Navigation.NavigationStack.Count > 1;

        EventHandler clickHandler = async (o, e) =>
        {
            var shouldNavigate = true;
            if (page is MyContentPage jpage)
            {
                shouldNavigate = jpage.ShouldNavigate();
            }
            if (shouldNavigate)
            {
                if (hasBackButton)
                {
                    await Element.Navigation.PopAsync();
                }
                else
                {
                    if (App.Current.MainPage is AppMainPage mainPage)
                    {
                        mainPage.IsPresented = !mainPage.IsPresented;
                    }                    
                }
            }
        };
        var menuButton = new UIImage("app_hamburger_icon.png");
        var backButton = new UIImage("back_icon_ios.png");
        var image = hasBackButton ? backButton : menuButton;
                
        var ratio =  (NavigationBar.Bounds.Height / 2) / image.Size.Height;
        var height = image.Size.Height * ratio;
        var width = image.Size.Width * ratio;
        var newSize = new CGSize(width, height);
        var img = image.Scale(newSize,0);
                
        var button = new UIBarButtonItem(img, UIBarButtonItemStyle.Plain, clickHandler);
                 
        var textAttr = new UIStringAttributes
        {
            Font = UIFont.FromName(UIFont.PreferredBody.Name, 25f)
        };

        button.SetTitleTextAttributes(textAttr, UIControlState.Normal);

        TopViewController.NavigationItem.SetLeftBarButtonItem(button, false);
    }
    return pushed;
}

```

---
#### Feedback Bot on 3/9/2023, 06:45 PM: 

(private comment, text removed)
#### John Ureke on 3/10/2023, 04:50 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by jsureke** [comment]
Please look in the original post for the sample code I already provided. It IS the "minimal sample project/solution or code samples which reproduce[s] the issue." This is a simple thing to duplicate, but here is the primary line which does not work in 17.6 Preview 1 when if the page being pushed is the top page in the stack:

 `TopViewController.NavigationItem.SetLeftBarButtonItem(button, false);`

When the page being pushed is the top of the navigation stack, then the button is not placed in the top left corner of the screen which means the click handler does not work.

If the page being pushed is not the top of the stack, then this works fine. It also worked fine for the first page in the stack in 17.5 Preview 6.

**Comment by jsureke** [comment]
Actually, that wasn't it.
In the derived ContentPage class, OnAppearing had the line 
NavigationPage.SetTitleView(this, CreateTitleView());
I guess replacing the title view does something to the NavigationItem.

**Comment by jsureke** [comment]
Either in the ContentPage constructor or in the XAML:

public partial class MyDisplayPage : ContentPage
{
   public MyDisplayPage()
   {
      NavigationPage.SetBackButtonTitle(this,"");
   }
}

<?xml version="1.0" encoding="utf-8"?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             NavigationPage.BackButtonTitle=""
             x:Class="MyProject.MyDisplayPage">


I always use my own BaseContentPage from which I create everything else.
That way, everything inherits all my base configuration:

namespace MyProject.Views
{
    public partial class BaseContentPage : ContentPage
    {
       public BaseContentPage()
       {
          NavigationPage.SetBackButtonTitle(this,"");
       }
    }
}

public partial class MyDisplayPage : BaseContentPage
{
   public MyDisplayPage()
   {
   }
}

<?xml version="1.0" encoding="utf-8"?>
<views:BaseContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
                               xmlns:views="clr-namespace:MyProject.Views"
             x:Class="MyProject.MyDisplayPage">

      
________________________________

**Comment by K0Ma19** [comment]
Thanks for the answer, but I use the shell and understand that this method is not suitable for me. I tried using `<Shell.BackButtonBehavior>
        <BackButtonBehavior TextOverride="Back" />   
    </Shell.BackButtonBehavior>` 
    @jsureke 
    
However, nothing happens. I found a workaround related to the picture, 
` <Shell.BackButtonBehavior>
        <BackButtonBehavior IconOverride ="back.png" />   
    </Shell.BackButtonBehavior>` 
    putting a picture it removes the text but then the problem will be on android and the back button is not displayed

**Comment by FM1973** [comment]
I can confirm, Setting the IconOverride works on IOS but not on Android.
Is there any workaround?

**Comment by lizharems** [comment]
@jsureke I tested this and was unable to repro this based on your explanation of your issue and attached code samples. I have attached the test project I used to attempt to recreate your issue. Let me know if you are seeing this issue in this project or if you have a full project/ solution with your issue that is different from what I implemented and I can look into it further.
[gtest.zip](https://github.com/dotnet/maui/files/12540526/gtest.zip)
